File: problems/3010/Solution.kt
class Solution {
    fun minimumCost(nums: IntArray): Int {
        var first = Int.MAX_VALUE
        var second = Int.MAX_VALUE
        for (i in 1 until nums.size) {
            val x = nums[i]
            if (x < first) { second = first; first = x }
            else if (x < second) second = x
        }
        return nums[0] + first + second
    }
}

File: problems/3010/solution.rs
impl Solution {
    pub fn minimum_cost(nums: Vec<i32>) -> i32 {
        let (mut first, mut second) = (i32::MAX, i32::MAX);
        for &x in &nums[1..] {
            if x < first { second = first; first = x; }
            else if x < second { second = x; }
        }
        nums[0] + first + second
    }
}
